src/AbortablePromiseCache.ts

```diff
@@ -69,12 +75,17 @@ export default class AbortablePromiseCache<T, U> {
     if (this.cache.get(key) === entry) this.cache.delete(key)
   }
 
-  fill(key: string, data: T, signal?: AbortSignal) {
+  fill(key: string, data: T, signal?: AbortSignal, statusCallback?: Function) {
     const aborter = new AggregateAbortController()
+    const statusReporter = new AggregateStatusReporter()
+    statusReporter.addCallback(statusCallback)
```

Comment:
What happens here if statusCallback is undefined?

Reply:
the AggregateStatusReporter has a default argument assignment to empty function. benefit of typescripting the module...makes sure things like this are ok